File: Programming Basics with Python/T07 - Nested Loops/lecture/04. Sum of Two Numbers.py
start_number = int(input())
end_number = int(input())
magic_number = int(input())
combination_number = 0
flag = False

for n1 in range(start_number, end_number + 1):
    for n2 in range(start_number, end_number + 1):
        combination_number += 1
        if n1 + n2 == magic_number:
            flag = True
            print(f"Combination N:{combination_number} ({n1} + {n2} = {magic_number})")
            break

    if flag:
        break

if n1 + n2 != magic_number:
    print(f"{combination_number} combinations - neither equals {magic_number}")

File: Programming Basics with Go/T07 - Nested Loops/lecture/04. Sum of Two Numbers.go
package main

import "fmt"

func main() {
	var startNumber, endNumber, magicNumber int
	fmt.Scanln(&startNumber)
	fmt.Scanln(&endNumber)
	fmt.Scanln(&magicNumber)
	var count int = 0
	var isFound bool = false
	for first := startNumber; first <= endNumber; first++ {
		for second := startNumber; second <= endNumber; second++ {
			count++
			if first + second == magicNumber {
				fmt.Printf("Combination N:%d (%d + %d = %d)", count, first, second, magicNumber)
				isFound = true
				break
			}
		}
		if isFound {
			break
		}
	}
	if !isFound {
		fmt.Printf("%d combinations - neither equals %d", count, magicNumber)
	}
}
